lib/LaTeXML/Package/diagbox.sty.ltxml

```diff
@@ -50,79 +50,96 @@ DefMacro('\diagbox OptionalKeyVals:diagbox {}{}', sub {
+        my $s     = $Bw + $Mw;
+        my $v     = $s * $t - $Aw * $Bh;
+        my $u     = $Aw * $Mh - $Mw * $Bh;
+        my $delta = ($u + $v) * ($u + $v) + 4 * $Aw * ($t - $Bh) * ($Mw * ($Bh - $Mh) - $Bw * $Mh);
+        $w = (($Bh < $t) && ($delta >= 0)
+          ? ($u + $v + sqrt($delta)) / ($t - $Bh) / 2
+          : 2 * (max($Aw, $Bw) + $Mw)) unless $w;
+        $h = (($Aw < $s) && ($delta >= 0)
+          ? ($u - $v - sqrt($delta)) / ($Aw - $s) / 2
+          : 2 * (max($Ah, $Bh) + $Mh)) unless $h; }
+      $w = roundto($w);
+      $h = roundto($h);
+      my ($wm, $hm) = ($w * 0.5, $h * 0.5);
       if ($dir eq 'NW') {
-        $Ay    = $Ah->ptValue;
-        $My    = $h->ptValue;
-        $Bx    = $w->subtract($Bw)->ptValue;
-        $By    = $h->ptValue;
-        $line1 = "$wwm,$hh $ww,0";
-        $line2 = "0,$hhm $ww,0"; }
+        $My    = $h - $Mh;
+        $Bx    = $w - $Bw;
+        $By    = $h - $Bh;
+        $line1 = "$wm,$h $w,0";
+        $line2 = "0,$hm $w,0"; }
       elsif ($dir eq 'SE') {
-        $Ay    = $Ah->ptValue;
-        $Mx    = $w->subtract($Mw)->ptValue;
-        $My    = $Mh->ptValue;
-        $Bx    = $w->subtract($Bw)->ptValue;
-        $By    = $h->ptValue;
-        $line1 = "0,$hh $ww,$hhm";
-        $line2 = "0,$hh $wwm,0"; }
+        $Mx    = $w - $Mw;
+        $Bx    = $w - $Bw;
+        $By    = $h - $Bh;
+        $line1 = "0,$h $w,$hm"; +
+          $line2 = "0,$h $wm,0"; }
       elsif ($dir eq 'SW') {
-        $Ay    = $h->ptValue;
-        $My    = $Mh->ptValue;
-        $Bx    = $w->subtract($Bw)->ptValue;
-        $By    = $Bh->ptValue;
-        $line1 = "0,$hhm $ww,$hh";
-        $line2 = "$wwm,0 $ww,$hh"; }
+        $Ay    = $h - $Ah;
+        $Bx    = $w - $Bw;
+        $line1 = "0,$hm $w,$h";
+        $line2 = "$wm,0 $w,$h"; }
       elsif ($dir eq 'NE') {
-        $Ay    = $h->ptValue;
-        $Mx    = $w->subtract($Mw)->ptValue;
-        $My    = $h->ptValue;
-        $Bx    = $w->subtract($Bw)->ptValue;
-        $By    = $Bh->ptValue;
-        $line1 = "0,0 $wwm,$hh";
-        $line2 = "0,0 $ww,$hhm"; } }
+        $Ay    = $h - $Ah;
+        $Mx    = $w - $Mw;
+        $My    = $h - $Mh;
+        $Bx    = $w - $Bw;
+        $line1 = "0,0 $wm,$h";
+        $line2 = "0,0 $w,$hm"; } }
     else {
-      $Ax = 0;
-      $Bx = $w->subtract($Bw)->ptValue;
+      $w  = 2 * $maxw unless $w;
+      $h  = $Ah + $Bh unless $h;
+      $w  = roundto($w);
+      $h  = roundto($h);
+      $Bx = $w - $Bw;
       if (($dir eq 'NW') || ($dir eq 'SE')) {
-        $Ay    = $Ah->ptValue;
-        $By    = $h->ptValue;
-        $line1 = "0,$hh $ww,0"; }
+        $By    = $h - $Bh;
+        $line1 = "0,$h $w,0"; }
       else {
-        $Ay    = $h->ptValue;
-        $By    = $Bh->ptValue;
-        $line1 = "0,0, $ww,$hh"; } }
-    $whatsit->setProperties(width => $w, height => $h,
+        $Ay    = $h - $Ah;
+        $line1 = "0,0, $w,$h"; } }
+    my $pxppt = Dimension('1pt')->pxValue(10);
```

Comment:
tempting to suggest hardcoding `$pxppt` to `1.38371` or such... Or do we expect this to change?

Reply:
Ah, the Elephant! Originally, I'd thought users would want to vary it according to target. But in the meantime, the web world has settled on an imaginary somewhat more constant "pixel".  And (my understanding of) SVG kinda forced my hand to use "pixels" more fundamentally in core.  So, yeah, maybe... later?